layouts/layouts/src/main/java/com/microsoft/device/dualscreen/layouts/FoldableFrameLayout.kt

```diff
@@ -59,10 +58,9 @@ open class FoldableFrameLayout @JvmOverloads constructor(
     }
 
     private fun registerWindowInfoFlow() {
-        val windowInfoRepository: WindowInfoRepository =
-            (context as Activity).windowInfoRepository()
         job = MainScope().launch {
```

Comment:
It's in the onDetachedFromWindow()

Reply:
Why not use like we use in other libs/samples?

```
lifecycleScope.launch(Dispatchers.Main) {
            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
                WindowInfoTracker.getOrCreate(this@MainActivity)
                    .windowLayoutInfo(this@MainActivity)
                    
                    ...
```

Reply:
We need to keep the reference to the coroutine job because we need to cancel it when the view is detached from the view tree.
It is not ideal to tie the coroutine to the activity lifecycle.  The view may be removed from the view tree (ex: with a removeView() call ) and the coroutine will be still running, it will deliver it's result. I this case  we end up changing the UI when the view is detached from the view tree.

Reply:
Thanks for the explanation Sorin :-)